### PR TITLE
Add check for triple

### PR DESCRIPTION
**Pull request by gautam1168**
For this issue: https://github.com/compiler-explorer/compiler-explorer/issues/4456

### CODE REVIEW

File: lib/buildenvsetup/ceconan-rust.ts [hunk 1]
```diff
@@ -48,11 +48,15 @@ export class BuildEnvSetupCeConanRustDirect extends BuildEnvSetupCeConanDirect {
     }
 
     getArchFromTriple(triple) {
```
Comment (matsjla) [inline lib/buildenvsetup/ceconan-rust.ts:50]:
```suggestion
    getArchFromTriple(triple: string): string {
```

**Review by matsjla — CHANGES_REQUESTED** [review]
Hey, thanks for the patch!

I think a better solution would be to check at the callsite of getArchFromTriple instead. In my opinion it doesn't really make sense that the `triple` should be allowed nullable/undefined in a function named getArchFromTriple

File: lib/buildenvsetup/ceconan-rust.ts (hunk 1 above)
Comment (gautam1168) [reply lib/buildenvsetup/ceconan-rust.ts:50]: yeah but the person who is on the issue said to check and return a blank string: https://github.com/compiler-explorer/compiler-explorer/issues/4456#issuecomment-1382856240

Comment (matsjla) [reply lib/buildenvsetup/ceconan-rust.ts:50]: Thoughts @partouf? I think it's certainly better if we keep getArchFromTriple clean of any bad inputs

**Review by matsjla — APPROVED** [review]

**Pull request merged**